pkg/actuators/machine/actuator.go

```diff
@@ -192,17 +193,21 @@ func (a *Actuator) CreateMachine(cluster *machinev1.Cluster, machine *machinev1.
 
 	// We explicitly do NOT want to remove stopped masters.
 	isMaster, err := a.isMaster(machine)
-	if err != nil {
-		return nil, a.handleMachineError(machine, apierrors.CreateMachine("error determining if machine is master: %v", err), createEventAction)
-	}
-	if !isMaster {
+	if err == nil && !isMaster {
 		// Prevent having a lot of stopped nodes sitting around.
 		err = removeStoppedMachine(machine, awsClient)
 		if err != nil {
 			glog.Errorf("unable to remove stopped machines: %v", err)
 			return nil, fmt.Errorf("unable to remove stopped nodes: %v", err)
 		}
 	}
+	// Unable to determine if a machine is a master machine.
```

Comment:
Would it make sense to move all of this closer to:

```go
isMaster, err := a.isMaster(machine)
```

Reply:
It's true it's first better to check if `err != nil` and then assume the error is nil rather than the other way around.